presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedSet.java

```diff
@@ -130,9 +134,29 @@ public void testBigintSimpleTypedSet()
         }
     }
 
+    @Test
+    public void testMemoryExceeded()
```

Comment:
hmm, i think it's just the default memory size...something really inefficient going on. setting Xms/Xmx 2g improves (but still can't do my 4M entry array, which seems a safe value to ensure we pass the 4MB underlying data limit)